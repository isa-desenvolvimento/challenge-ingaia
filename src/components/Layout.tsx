import TranslationProvider from 'locales/TranslationProvider'
import React from 'react'
import { ChildrenProps } from 'types/components'

export default function Layout({ children }: ChildrenProps) {
  return (
    <TranslationProvider>
      <main>{children}</main>
    </TranslationProvider>
  )
}
