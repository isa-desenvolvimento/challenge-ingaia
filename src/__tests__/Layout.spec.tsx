/* eslint-disable global-require */
import { render } from '@testing-library/react';
import Layout from '../components/Layout';

jest.mock('next/router', () => require('next-router-mock'));

describe('<Layout />', () => {
  it('should render component Layout children', () => {
    const { getByRole } = render(
      <Layout>
        <h1>Test Layout</h1>
      </Layout>,
    );
    expect(getByRole('heading', { name: /test Layout/i })).toBeInTheDocument();
  });

  it('should render snapshot', () => {
    const { container } = render(
      <Layout>
        <h1>Test Layout</h1>
      </Layout>,
    );
    expect(container).toMatchSnapshot();
  });
});
