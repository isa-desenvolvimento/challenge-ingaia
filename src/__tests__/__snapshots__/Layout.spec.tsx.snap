// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render snapshot 1`] = `
<div>
  <nav
    class="container"
  >
    <div
      class="logo"
      data-testid="img_nav"
    />
    <form
      class="inputContainer"
      data-testid="frm_nav"
    >
      <input
        data-testid="inp_nav"
        placeholder="Nunca dejes de buscar"
        type="text"
      />
      <button
        class="search"
        data-testid="btn_nav"
        type="submit"
      >
         
      </button>
    </form>
  </nav>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1>
        Test Layout
      </h1>
    </main>
  </div>
</div>
`;
